src/Microsoft.Extensions.DependencyInjection.Specification.Tests/DependencyInjectionSpecificationTests.cs

```diff
@@ -0,0 +1,542 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection.Specification.Fakes;
+using Xunit;
+
+namespace Microsoft.Extensions.DependencyInjection.Specification
+{
+    public abstract partial class DependencyInjectionSpecificationTests
+    {
+        protected abstract IServiceProvider CreateServiceProvider(IServiceCollection serviceCollection);
+
+        [Fact]
+        public void ServicesRegisteredWithImplementationTypeCanBeResolved()
+        {
+            // Arrange
+            var collection = new ServiceCollection();
+            collection.AddTransient(typeof(IFakeService), typeof(FakeService));
+            var container = CreateServiceProvider(collection);
+
+            // Act
+            var service = container.GetService<IFakeService>();
+
+            // Assert
+            Assert.NotNull(service);
+            Assert.IsType<FakeService>(service);
+        }
+
+        [Fact]
+        public void ServicesRegisteredWithImplementationType_ReturnDifferentInstancesPerResolution_ForTransientServices()
```

Comment:
I think this test should be expanded to verify that transients are unique when resolved from a different scope as well. Not just the root provider. Either that or add a different test.